src/main/java/org/apache/sling/feature/scanner/impl/ContentPackageScanner.java

```diff
@@ -238,13 +242,16 @@ private ArtifactId extractArtifactId(final File tempDir, final File bundleFile)
         final File toDir = new File(tempDir, bundleFile.getName());
         toDir.mkdirs();
 
-        try (final ZipInputStream zis = new ZipInputStream(new FileInputStream(bundleFile)) ) {
-            boolean done = false;
-            while ( !done ) {
-                final ZipEntry entry = zis.getNextEntry();
-                if ( entry == null ) {
-                    done = true;
-                } else {
+        ZipFile zipFile = null;
```

Comment:
Same as above - try to use the try() blocks and only open the streams if you need them (and don't forget to close)